src/commands/postDeploy/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,176 @@
+            }
+
+            // update the new end time to be now
+            // tslint:disable-next-line: no-unsafe-any
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(context, requestOptions, deployResultTime, node.root.client.fullName);
+
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        if (showOutput) {
+            // this detector has more valuable information to show in the portal than 'LinuxContainerStartFailure'
+            const linuxAppDownId: string = 'LinuxAppDown';
+            await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+            await openInPortal(node.root, `${node.root.client.id}/troubleshoot`, { queryPrefix: `websitesextension_ext=asd.featurePath%3Ddetectors%2F${linuxAppDownId}` });
+            context.telemetry.properties.didClick = 'true';
+        }
+    });
+}
+
+async function getLinuxDetectorError(context: IActionContext, requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
+    const detectorResponse: string = await requestUtils.sendRequest<string>(requestOptions);
+    const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
+
+    const datasets: detectorDataset[] = responseJson.properties.dataset;
+    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
+    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+
+    // if we can't find the timestamp, exit and try again
+    if (!timestampTable) {
+        return undefined;
+    }
+
+    const timestamp = new Date(getValuesByColumnName(timestampTable, ColumnName.value)[0]);
+    if (validateTimestamp(timestamp, deployResultTime)) {
+
+        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
+        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
+
+        // if the criticalError table or failureCounts aren't defined, we can't display anything to the user
+        if (!criticalErrorTable || !failureCountTable) {
+            return undefined;
+        }
+
+        const errorMessages: string[] = getValuesByColumnName(criticalErrorTable, ColumnName.value);
+        const failureCount: string = getValuesByColumnName(failureCountTable, ColumnName.failureCount)[0];
+
+        context.telemetry.properties.errorMessages = JSON.stringify(errorMessages);
+        context.telemetry.properties.failureCount = failureCount;
+
+        return `Critical insights found for "${siteName}": ${failureCount} as of ${new Date(timestamp).toLocaleString()}. Critical insight preview: "${errorMessages[0]}". Click here to access "App Service Diagnostics" for recommendations.`;
```

Comment:
Can I see a screenshot of this

Reply:
![image](https://user-images.githubusercontent.com/5290572/64659046-ae369280-d3ee-11e9-8d40-3779e70dd999.png)